### PR TITLE
Support pathlib.Path in write_if_different and import_file

### PR DESCRIPTION
**Pull request by astrofrog**
Replacement for https://github.com/astropy/extension-helpers/pull/83/ which includes tests

### CODE REVIEW

**Comment by neutrinoceros** [comment]
@astrofrog Looks like we both worked on this at the very same time (#85) !
My PR looks simpler though, what do you think ?

**Comment by astrofrog** [comment]
@neutrinoceros - we need to keep supporting str input for those functions, hence some of the changes I made here, and I've also been able to simplify a little of the code using read_bytes/write_bytes. The tests here also check both str and Path work, so I'd prefer to keep #84 and close #85 if you agree.

**Comment by neutrinoceros** [comment]
> The tests here also check both str and Path work, so I'd prefer to keep https://github.com/astropy/extension-helpers/pull/84 and close https://github.com/astropy/extension-helpers/pull/85 if you agree.

I didn't drop support for `str` in my PR, and I also have tests for both path flavors, so both PRs are pretty much equivalent there, but yours does have additional documentation change so it should probably be favored, yes.
My one critic here is that you're changing the name of an argument that's keyword-allowed, which seems like an unwarranted breaking change.

**Comment by astrofrog** [comment]
> I didn't drop support for str in my PR

Sorry I wrote my comment before you pushed some additional changes :sweat_smile: - race conditions!

File: extension_helpers/_utils.py [hunk 1]
```diff
@@ -77,7 +78,7 @@ def walk_skip_hidden(top, onerror=None, followlinks=False):
         yield root, dirs, files
 
 
-def write_if_different(filename, data):
```
Comment (neutrinoceros) [inline extension_helpers/_utils.py:80]: This is the breaking change I was talking about. Can you please revert it ? Beyond that, happy to approve this PR.

**Review by neutrinoceros — CHANGES_REQUESTED** [review]

File: extension_helpers/_utils.py (hunk 1 above)
Comment (astrofrog) [reply extension_helpers/_utils.py]: Done and pushed

**Review by neutrinoceros — APPROVED** [review]
Absolutely. Thank you !

**Pull request merged**